scripts/compare_benchmarks.py

```diff
@@ -84,74 +144,195 @@ def calculate_and_format_p_value(old, new):
         table_data.append(unsuccessful_info_colored)
 
-table_data.append(["geometric mean", "", "", "", "", format_diff(geometric_mean(diffs)), ""])
+table_data.append(
+    [
+        "Sum",
+        f"{(total_runtime_old / 1e6):>7.1f}",
+        f"{(total_runtime_new / 1e6):>7.1f}",
+        color_diff(total_runtime_new / total_runtime_old, True) + " ",
+    ]
+)
+table_data.append(["Geomean", "", "", "", "", "", color_diff(geometric_mean(diffs_throughput)) + " "])
 
 table = AsciiTable(table_data)
-table.justify_columns[6] = "right"
+for column_index in range(1, len(table_data[0])):  # all columns justified to right, except for item name
+    table.justify_columns[column_index] = "right"
+
+result = str(table.table)
+
+new_result = ""
+lines = result.splitlines()
+
+
+# Double the vertical line between the item names and the two major measurements.
+lines = double_vertical_separators(lines, [1, 4])
+
+# As the used terminaltables module does not support cells that span multiple columns, we do that manually for latency
+# and throughput in the header. We used two place holders that are narrow enough to not grow the column any wider than
+# necessary for the actual values. After manually changing the column title to span two column, we replace the place
+# holder with the actual full descriptions.
+for (placeholder, final) in [
+    ("$thrghpt", "Throughput (iter/s)"),
+    ("$latency", "Latency (ms/iter)"),
+]:
+    header_strings = lines[1].split("|")
+    for column_id, text in enumerate(header_strings):
+        if placeholder in text:
+            title_column = header_strings[column_id]
+            unit_column = header_strings[column_id + 1]
+            previous_length = len(title_column) + len(unit_column) + 1
+            new_title = f" {final} ".ljust(previous_length, " ")
+            lines[1] = "|".join(header_strings[:column_id] + [new_title] + header_strings[column_id + 2:])
+
+
+# Swap second line of header with automatically added separator. Terminaltables does not support multi-line headers. So
+# we have the second header line as part of the results after a separator line. We need to swap these.
+lines[2], lines[3] = lines[3], lines[2]
+for (line_number, line) in enumerate(lines):
+    if line_number == len(table_data):
+        # Add another separation between benchmark items and aggregates
+        new_result += lines[-1] + "\n"
+
+    new_result += line + "\n"
+
+
+# In case the runs for the executed benchmark have been cut or the number of runs was insufficient for the p-value
+# calcution, we add notes to the end of the table.
+if add_note_for_capped_runs or add_note_for_insufficient_pvalue_runs:
+    first_column_width = len(lines[1].split("|")[1])
+    width_for_note = len(lines[0]) - first_column_width - 5  # 5 for seperators and spaces
+    if add_note_for_capped_runs:
+        note = "˄" + f' Execution stopped at {new_data["context"]["max_runs"]} runs'
```

Comment:
Why new_data?

Reply:
Good point. Since we have the overview table of the context anyways, simple have a note stating "Execution stopped due to max runs reached."?

Reply:
Yepp.